### ISSUE TITLE
Support "Add to App" flutter integrations in native apps (Don't start SentrySDK if it is already running)

### ISSUE THREAD

**Issue by charlesbensonsonos** [getsentry/sentry-dart#544]
TL:DR - Add to App use cases where Flutter is integrated into an existing native app can have their configured SentrySDK instance stopped and 'overridden' by sentry-dart, which starts its own native SentrySDK instance. (At least for iOS)

## Problem Statement
Our app is a native platform built for Android & iOS that integrates flutter for some parts of the UI of the app, but is dominantly a natively developed app. Flutter calls this the "Add to App" experience: https://flutter.dev/docs/development/add-to-app

After integrating sentry-dart into our native iOS app which was already using sentry-cocoa, we discovered that the flutter library was creating a new SentrySDK.start() instance, effectively stopping the previous instance we created in the native side of the app. This startup behavior is not necessary since we already had an instance of SentrySDK running. This code was temporarily commented out for our implementation tests, and worked just fine using our already started instance for the native app.

Since we are using "native" dominant apps, the apps each have their own project buckets on Sentry.io, one for the android app, one for the iOS app, which allows for the flutter exception events to route to the native platform project buckets, preventing the need for additional tagging of the platform and sentry library in the flutter crash. This code was also commented out.

The native app is also starting and stopping flutter as it moves between native ui and flutter ui. We wanted to keep Sentry running after the 'flutter side' was closed, so we commented out the SentrySDK.close() code as well.

Since we have a native instance of Sentry running, we are using the native DSN, and do not need to set a DSN for flutter, however SentryFlutter.init() has a requirement for options to be set. For now, we are feeding in an empty string for dsn.

```dart
await SentryFlutter.init(
  (options) => options.dsn = '' // No DSN needed as we are relying on the native SDK instance which already has a DSN assigned. 
);
```

Edited (commented out) function code in SentryFlutterPluginApple.swift
```swift
    private func initNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let arguments = call.arguments as? [String: Any], !arguments.isEmpty else {
            print("Arguments is null or empty")
            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil))
            return
        }
// Sentry SDK is already running and configured in our app. Comment out this call to start().
//        SentrySDK.start { options in
//            self.updateOptions(arguments: arguments, options: options)
//
//            self.sentryOptions = options
//
// No need to configure extra tags with platform details as the native SDK is already configured for this data
//            // note : for now, in sentry-cocoa, beforeSend is not called before captureEnvelope
//            options.beforeSend = { event in
//                self.setEventOriginTag(event: event)
//
//                if var sdk = event.sdk, self.isValidSdk(sdk: sdk) {
//                    if let packages = arguments["packages"] as? [String] {
//                        if var sdkPackages = sdk["packages"] as? [String] {
//                            sdk["packages"] = sdkPackages.append(contentsOf: packages)
//                        } else {
//                            sdk["packages"] = packages
//                        }
//                    }
//
//                    if let integrations = arguments["integrations"] as? [String] {
//                        if var sdkIntegrations = sdk["integrations"] as? [String] {
//                            sdk["integrations"] = sdkIntegrations.append(contentsOf: integrations)
//                        } else {
//                            sdk["integrations"] = integrations
//                        }
//                    }
//                    event.sdk = sdk
//                }
//
//                return event
//            }
//        }
//
        // checking enableAutoSessionTracking is actually not necessary, but we'd spare the sent bits.
        if didReceiveDidBecomeActiveNotification {
            // we send a SentryHybridSdkDidBecomeActive to the Sentry Cocoa SDK, so the SDK will mimics
            // the didBecomeActiveNotification notification and start a session if not yet.
            NotificationCenter.default.post(name: Notification.Name("SentryHybridSdkDidBecomeActive"), object: nil)
            // we reset the flag for the sake of correctness
            didReceiveDidBecomeActiveNotification = false
        }

        result("")
    }

    private func closeNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        // No need to close Sentry here. This is being controlled by the native app. 
        //SentrySDK.close()
        result("")
    }
```

## Proposal
SentryFlutter.init() in Dart could support optional options, or a flag within the options to say that the execution is for the "Add-to-App" experience, where a DSN is not necessary.

If SentrySDK is already running natively, do not shut it down and create a new instance for Flutter, instead use the already running instance being used by the native app. 

This will help support the add-to-app bits of Flutter pretty quickly, but I'm not sure if this is the best design. Any thoughts on the right way to support this?

**Comment by ueman** [comment]
Hey, thanks for this throughout issue description. I understand that it can be useful to support that scenario.

This issue is kinda related https://github.com/getsentry/sentry-dart/issues/265

**Comment by marandaneto** [comment]
yep I'd say so its kinda related to it, a workaround might be removing the `NativeSdkIntegration` from the `integrations` list or using the https://pub.dev/packages/sentry instead of sentry_flutter

**Comment by marandaneto** [comment]
The RN SDK has an option called [enableNative](https://github.com/getsentry/sentry-react-native/blob/1b53983728d4f27c25aff242d34616e8200de017/src/js/options.ts#L13-L19).
We could add support for it and based on that, we don't call the Native bridge for initing the SDK.
It's needed for https://github.com/getsentry/sentry-dart/issues/265 as well

**Comment by marandaneto** [comment]
Dup of #265
